src/components/autocomplete/js/autocompleteController.js

```diff
@@ -582,14 +587,20 @@ function MdAutocompleteCtrl ($scope, $element, $mdUtil, $mdConstant, $mdTheming,
    * @param index
    */
   function select (index) {
+    var selectedItem = (typeof ctrl.matches[ index ] !== 'undefined') ? ctrl.matches[ index ] : null;
```

Comment:
This is a bit long for one line. Can you perhaps refactor this into a simple function whose name describes a bit what is happening?